src/metabase/lib/metadata/protocols.cljc

```diff
@@ -21,57 +20,116 @@
 
   This protocol is pretty limited at this point; in the future, we'll probably want to add:
 
-  - methods to facilitate more fine-grained access in the future, such as methods that let us fetch more limited info
-    about a large number of objects, then other methods to let you fetch more complete information for a single object
-    once you make a selection. E.g. the first method might be used to power a list of Tables to choose from to join
-    against, then once you choose a Table we hit the second method to get more info about it
-
   - methods for searching for Tables or Fields matching some string
 
   - paging, so if you have 10k Tables we don't do crazy requests that fetch them all at once
 
-  But things like fine grained-access and search can be implemented with just the basic `tables` and `fields`
-  methods for the time being, so we don't need to figure out all of this stuff upfront."
+  For all of these methods: if no matching object can be found, you should generally return `nil` rather than throwing
+  an Exception. Let [[metabase.lib.metadata]] worry about throwing exceptions."
   (database [this]
     "Metadata about the Database we're querying. Should match the [[metabase.lib.metadata/DatabaseMetadata]] schema.
   This includes important info such as the supported `:features` and the like.")
+
+  (table [this table-id]
+    "Return metadata for a specific Table. Metadata should satisfy [[metabase.lib.metadata/TableMetadata]].")
+
+  (field [this field-id]
+    "Return metadata for a specific Field. Metadata should satisfy [[metabase.lib.metadata/ColumnMetadata]].")
+
+  (metric [this metric-id]
+    "Return metadata for a particular capital-M Metric, i.e. something from the `metric` table in the application
+    database. Metadata should match [[metabase.lib.metadata/MetricMetadata]].")
+
+  (segment [this segment-id]
+    "Return metadata for a particular captial-S Segment, i.e. something from the `segment` table in the application
+    database. Metadata should match [[metabase.lib.metadata/SegmentMetadata]]." )
+
+  ;; Card not yet implemented! I commented it out for now with the intention that we'll probably want to uncomment it
+  ;; out and fully implement it in the very near future.
+
+  ;; (card [this card-id]
```

Comment:
I was going to implement a `card` metadata method but I decided this PR had enough going on already, I commented it out for now but should do it in the near future